bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -115,4 +129,70 @@
      */
     public void runNow(String ruleUID);
 
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a given scope or {@code null} scope.
+     * @param scope to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasScope(String scope) {
+        if (null == scope) {
+            return r -> null == r.getScope();
+        } else {
+            return r -> scope.equals(r.getScope());
+        }
+    }
+
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for any of the given scopes and even {@code null} scope.
+     * @param scopes to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasAnyScope(String... scopes) {
+        HashSet<String> scopeSet = new HashSet<String>(scopes.length);
+        for(final String scope : scopes) {
+            scopeSet.add(scope);
+        }
+
+        // this will even work for null scopes
+        return r-> scopeSet.contains(r.getScope());
+    }
+
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a all given tags or {@link Rule}s without tags.
+     * All given tags must match, (the matched {@code Rule} might contain more).
+     * @param tags to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasAllTags(Collection<String> tags) {
+        if (null == tags || tags.isEmpty()) {
+            // everything without a tas is matching
+            // Ruler.getTags() is never null
+            return r -> r.getTags().isEmpty();
+        } else {
+            Set<String> tagSet = new HashSet<String>(tags);
+
+            // everything containing _all_ _given_ tags is matching (Rule might might have more tags than the given set)
+            return r -> r.getTags().containsAll(tagSet);
+        }
+    }
+
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a any of given tags or {@link Rule}s without tags.
+     * @param tags to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasAnyTags(Collection<String> tags) {
+        if (null == tags || tags.isEmpty()) {
```

Comment:
Same as above. IMHO the interface should not be filled with utility methods. This functionality can be achieved by multiple calling getByTag() method or again you can use getAll(Predicate) method in your application.